reakit/reakit
# feat: Add `shift` option to `useCompositeState`

This PR promotes the `unstable_angular` option on `useCompositeState` to a stable version and renames it to `shift`.

As the docs imply, this feature consists of shiftting focus to the first enabled item before an empty or disabled cell when moving up or down in a composite grid widget.

**Screenshots**

In the example below, the default behavior would move focus to the "Gallery" item. With `shift` enabled, it moves focus to "Verse"

<img width="340" alt="Screenshot exemplifying the statement above" src="https://user-images.githubusercontent.com/3068563/85659605-15c28380-b68b-11ea-9c74-9a1f4dd1d870.png">

**How to test?**

Check https://deploy-preview-727--reakit.netlify.app/storybook/?path=/story/examples--composite-shift and the tests for that story.

**Does this PR introduce breaking changes?**

Not technically a breaking change as this is updating an experimental feature. But this will break code of people using the `unstable_angular` option.

```diff
- const composite = useCompositeState({ unstable_angular: true });
+ const composite = useCompositeState({ shift: true });
```
